src/btree/bt_huffman.c

```diff
@@ -128,28 +128,67 @@ static const struct __wt_huffman_table __wt_huffman_nytenglish[] = {
 static int __wt_huffman_read(WT_SESSION_IMPL *,
     WT_CONFIG_ITEM *, struct __wt_huffman_table **, u_int *, u_int *);
 
-#define	WT_HUFFMAN_CONFIG_VALID(str, len)				\
-	(WT_STRING_MATCH("english", (str), (len)) ||			\
-	    WT_PREFIX_MATCH((str), "utf8") || WT_PREFIX_MATCH((str), "utf16"))
-
 /*
- * __btree_huffman_config --
- *	Verify the key or value strings passed in.
+ * __huffman_confchk_file --
+ *	Check for a Huffman configuration file and return the file name.
  */
 static int
-__btree_huffman_config(WT_SESSION_IMPL *session,
-    WT_CONFIG_ITEM *key_conf, WT_CONFIG_ITEM *value_conf)
+__huffman_confchk_file(
+    WT_SESSION_IMPL *session, WT_CONFIG_ITEM *v, int *is_utf8p, FILE **fpp)
 {
-	if (key_conf->len != 0 &&
-	    !WT_HUFFMAN_CONFIG_VALID(key_conf->str, key_conf->len))
-		WT_RET_MSG(
-		    session, EINVAL, "illegal Huffman key configuration");
-	if (value_conf->len != 0 &&
-	    !WT_HUFFMAN_CONFIG_VALID(value_conf->str, value_conf->len))
-		WT_RET_MSG(
-		    session, EINVAL, "illegal Huffman value configuration");
+	FILE *fp;
+	size_t len;
+	char *fname;
+
+	/* Look for a prefix and file name. */
+	len = 0;
+	if (WT_PREFIX_MATCH(v->str, "utf8")) {
+		if (is_utf8p != NULL)
+			*is_utf8p = 1;
+		len = strlen("utf8");
+	} else if (WT_PREFIX_MATCH(v->str, "utf16")) {
+		if (is_utf8p != NULL)
+			*is_utf8p = 0;
```

Comment:
These two lines should move outside the conditional up with `len=0;` so that we always initialize `is_utf8p` to something.